insights/components/openstack.py

```diff
@@ -0,0 +1,27 @@
+"""
+IsOpenStackCompute
+==================
+
+The ``IsOpenStackCompute`` component uses ``PsAuxcww`` parser to determine
+OpenStack Compute node. It checks if 'nova-compute' process exist, if not raises
+``SkipComponent`` so that the dependent component will not fire. Can be added as
+a dependency of a parser so that the parser only fires if the
+``IsIsOpenStackCompute`` dependency is met.
+"""
+from insights.core.plugins import component
+from insights.parsers.ps import PsAuxcww
+from insights.core.dr import SkipComponent
+
+
+@component(PsAuxcww)
+class IsOpenStackCompute(object):
+    """The ``IsOpenStackCompute`` component uses ``PsAuxcww`` parser to determine
+    OpenStack Compute node. It checks if ``nova-compute`` process exist, if not
+    raises ``SkipComponent``.
+
+    Raises:
+        SkipComponent: When ``nova-compute`` process does not exist.
+    """
+    def __init__(self, ps):
+        if ('nova-compute' not in ps.running) and ('nova-conductor' in ps.running):
```

Comment:
I don't think this check follows the above description of the Component.
Think about there is not `'nova-compute'` and `'nova-conductor'` processes, it must not an OpenStackCompute node, right? but in that case, the IsOpenStackCompute component would be `NOT None` which will cause the Parser getting fired.

Reply:
@xiangce The logic is aligned with the description but we have to think as a "negation".
The `SkipComponent` will be raised if `nova-compute` is not in the output of `ps` which means the node is not running compute services, hence not a compute node.

To support this, I've covered three tests in `components/tests/test_openstack.py` where:

1. `test_compute_node`: `nova-compute` exist in `ps`
2. `test_controller_node`: The controller node will not have `nova-compute` process running
3. `test_director_node`: The director node in OpenStack acts is all in node(compute + controller). `IsOpenStackCompute` will not throws an exception here.

Reply:
For a normal RHEL-host, when there is no 'nova-compute' or 'nova-conductor' process running, The `IsOpenStackCompute` will be something Truthy and cause this Parser getting fired incorrectly.

Reply:
@xiangce Removed unwanted `nova-condutor` check.